src/main/java/com/goodtrip/goodtripserver/database/repositories/TripRepository.java

```diff
@@ -0,0 +1,96 @@
+package com.goodtrip.goodtripserver.database.repositories;
+
+import com.goodtrip.goodtripserver.database.models.CountryVisit;
+import com.goodtrip.goodtripserver.database.models.Note;
+import com.goodtrip.goodtripserver.database.models.Trip;
+import com.goodtrip.goodtripserver.database.models.TripState;
+import jakarta.annotation.Nullable;
+import org.springframework.stereotype.Repository;
+
+import java.sql.Date;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public interface TripRepository {
```

Comment:
Hint, if you want, you can in this Interface specify only your custom methods and extend this interface with JPARepository  or CrudRepository, which provide basic operations and Spring automatically creates implementations for them

Reply:
I found out about it after I wrote this code, so, I want to refactor this code and extend interface with JPARepository, but I had some troubles with beans, so I thought that I'll do this later.